File: app/src/main/java/com/viafourasample/src/activities/splash/SplashActivityViewModel.java
package com.viafourasample.src.activities.splash;

import android.os.CountDownTimer;

public class SplashActivityViewModel {
    private SplashActivityCallback splashActivityCallback;

    public SplashActivityViewModel() {
        new CountDownTimer(5000, 1000) {

            public void onTick(long millisUntilFinished) {

            }

            public void onFinish() {
                splashActivityCallback.onLoadingCompleted();
            }

        }.start();
    }

    public void setSplashActivityCallback(SplashActivityCallback splashActivityCallback) {
        this.splashActivityCallback = splashActivityCallback;
    }

    public interface SplashActivityCallback {
        void onLoadingCompleted();
    }
}

File: app/src/main/java/com/viafourasample/src/activities/splash/SplashActivityViewModel.kt
package com.viafourasample.src.activities.splash

import android.os.CountDownTimer

class SplashActivityViewModel {
    var splashActivityCallback: SplashActivityCallback? = null

    init {
        object : CountDownTimer(5000, 1000) {
            override fun onTick(millisUntilFinished: Long) {
            }

            override fun onFinish() {
                splashActivityCallback?.onLoadingCompleted()
            }
        }.start()
    }

    fun interface SplashActivityCallback {
        fun onLoadingCompleted()
    }
}
